src/pages/resultsView/survival/SurvivalChart.spec.tsx

```diff
@@ -0,0 +1,184 @@
+            assert.deepEqual(getLineData(examplePatientSurvivals, exampleEstimates), [
+                {x: 0, y: 100},
+                {x: 5.09, y: 100},
+                {x: 0.09, y: 80},
+                {x: 0, y: 80},
+                {x: 63.83, y: 53.3333333333333333},
+                {x: 0.13, y: 53.3333333333333333},
+                {x: 182.19, y: 0}
+            ]);
+        });
+    });
+
+    describe("#getScatterData()", () => {
+        it("returns empty list for empty list", () => {
+            assert.deepEqual(getScatterData([], []), []);
+        });
+
+        it("returns correct scatter data for the example data", () => {
+            assert.deepEqual(getScatterData(examplePatientSurvivals, exampleEstimates), [
+                {x: 5.09, y: 100, patientId: "TCGA-OR-A5J1", studyId: "acc_tcga", status: false},
+                {x: 0.09, y: 80, patientId: "TCGA-OR-A5J2", studyId: "acc_tcga", status: true},
+                {x: 0, y: 80, patientId: "TCGA-OR-A5J3", studyId: "acc_tcga", status: false},
+                {x: 63.83, y: 53.3333333333333333, patientId: "TCGA-2F-A9KO", studyId: "blca_tcga", status: true},
+                {x: 0.13, y: 53.3333333333333333, patientId: "TCGA-2F-A9KP", studyId: "blca_tcga", status: false},
+                {x: 182.19, y: 0, patientId: "TCGA-2F-A9KQ", studyId: "blca_tcga", status: true}
+            ]);
+        });
+    });
+
+    describe("#getScatterDataWithOpacity()", () => {
+        it("returns empty list for empty list", () => {
+            assert.deepEqual(getScatterDataWithOpacity([], []), []);
+        });
+
+        it("returns correct scatter data with opacity for the example data", () => {
+            assert.deepEqual(getScatterDataWithOpacity(examplePatientSurvivals, exampleEstimates), [
+                {x: 5.09, y: 100, patientId: "TCGA-OR-A5J1", studyId: "acc_tcga", status: false, opacity: 1},
+                {x: 0.09, y: 80, patientId: "TCGA-OR-A5J2", studyId: "acc_tcga", status: true, opacity: 0},
+                {x: 0, y: 80, patientId: "TCGA-OR-A5J3", studyId: "acc_tcga", status: false, opacity: 1},
+                {x: 63.83, y: 53.3333333333333333, patientId: "TCGA-2F-A9KO", studyId: "blca_tcga", status: true, opacity: 0},
+                {x: 0.13, y: 53.3333333333333333, patientId: "TCGA-2F-A9KP", studyId: "blca_tcga", status: false, opacity: 1},
+                {x: 182.19, y: 0, patientId: "TCGA-2F-A9KQ", studyId: "blca_tcga", status: true, opacity: 0}
+            ]);
+        });
+    });
+
+    describe("#getStats()", () => {
+        it("returns [0, 0, \"NA\"] for empty list", () => {
+            assert.deepEqual(getStats([], []), [0, 0, "NA"]);
+        });
+
+        it("returns correct stats for the example data", () => {
+            assert.deepEqual(getStats(examplePatientSurvivals, exampleEstimates), [6, 3, "182.19"]);
+        });
+    });
+
+    describe("#calculateLogRank()", () => {
+        it("returns NaN for empty list", () => {
```

Comment:
this seems like a good example for discussion.  there is some complicated code inside this function that would require many more assertions to cover.   we need to figure out how to define "covered".